File: solutions/Leetcode_970/leetcode_970.py
class Solution1:
    def powerful_integers(self, x, y, bound):
        s = set()
        i = 1
        while i <= bound:
            j = 1
            while i + j <= bound:
                s.add(i+j)
                if y == 1:
                    break
                j *= y
            if x == 1:
                break
            i *= x
        return list(s)


class SolutionPythonicOf1:
    def powerful_integers(self, x, y, bound):
        power_x = {x**i for i in range(20) if x**i <= bound}
        power_y = {y**j for j in range(20) if y**j <= bound}
        return list({xx + yy for xx in power_x for yy in power_y if xx+yy <= bound})

File: solutions/Leetcode_970/leetcode_970.cc
#include <unordered_set>
#include <vector>

using namespace std;

class Solution1
{
public:
    vector<int> powerfulIntegers(int x, int y, int bound)
    {
        unordered_set<int> set;
        for (int i = 1; i <= bound; i *= x)
        {
            for (int j = 1; i + j <= bound; j *= y)
            {
                set.insert(i + j);
                if (y == 1)
                    break;
            }
            if (x == 1)
                break;
        }
        return vector<int>(set.begin(), set.end());
    }
};